### PR TITLE
Tweak: Import All command should skip the Plugins screen and start import process[ED-6510]

### PR DESCRIPTION
**Pull request by davseve**

### CODE REVIEW

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Review by yossi-elementor — CHANGES_REQUESTED** [review]

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+				navigate( '/import/plugins-activation' );
+			}
+		};
+
+		useEffect( () => {
+			handleRequiredPlugins();
+		}, [ plugins, missing ] );
+
+	return (
+		<span>.</span>
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:31]: This file has been deleted

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import { ImportContext } from '../../../../../context/import-context/import-context-provider';
+import usePlugins from '../../../../../hooks/use-plugins';
+import usePluginsData from '../../../../../hooks/use-plugins-data';
+import useImportPluginsData from '../../hooks/use-import-plugins-data';
+import { useNavigate } from '@reach/router';
+
+export default function ImportKitLibraryPlugins() {
+	const importContext = useContext( ImportContext ),
+		plugins = importContext.data.plugins || {},
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:11]: This file has been deleted

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		handleRequiredPlugins = () => {
+			if ( missing?.length ) {
+				// Saving globally the plugins data that the kit requires in order to work properly.
+				importContext.dispatch( { type: 'SET_PLUGINS', payload: missing } );
+				navigate( '/import/plugins-activation' );
+			}
+		};
+
+		useEffect( () => {
+			handleRequiredPlugins();
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:27]: This file has been deleted

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		}, [ plugins, missing ] );
+
+	return (
+		<span>.</span>
+	);
+}
+
+ImportKitLibraryPlugins.propTypes = {
+	plugins: PropTypes.array,
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:36]: This file has been deleted
This is probably a leftover of previous changes

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+};
+
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:38]: This file has been deleted

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	const importContext = useContext( ImportContext ),
+		plugins = importContext.data.plugins || {},
+		navigate = useNavigate(),
+		{ response } = usePlugins(),
+		//all installed plugins
+		{ pluginsData } = usePluginsData( response.data ),
+		{ importPluginsData } = useImportPluginsData( plugins, pluginsData ),
+		{ missing } = importPluginsData || {},
+		handleRequiredPlugins = () => {
+			if ( missing?.length ) {
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:19]: This file has been deleted

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/hooks/use-import-kit-library-plugins.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { useContext, useEffect } from 'react';
+import usePlugins from '../../../../hooks/use-plugins';
+import usePluginsData from '../../../../hooks/use-plugins-data';
+import useImportPluginsData from './use-import-plugins-data';
+import { useNavigate } from '@reach/router';
+import { ImportContext } from '../../../../context/import-context/import-context-provider';
+
+export default function UseImportKitLibraryPlugins() {
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/hooks/use-import-kit-library-plugins.js:8]: This file has been deleted, But I will pay more attention to it in the future.

File: core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js [hunk 8]
```diff
@@ -51,7 +67,9 @@ export default function ImportProcess() {
 		if ( referrer ) {
 			sharedContext.dispatch( { type: 'SET_REFERRER', payload: referrer } );
 		}
-
+		if ( actionType ) {
+			importContext.dispatch( { type: 'SET_ACTIONTYPE', payload: actionType } );
+		}
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js:72]: Done

File: core/app/modules/import-export/assets/js/context/import-context/import-context-reducer.js [hunk 9]
```diff
@@ -20,6 +20,10 @@ export const reducer = ( state, { type, payload } ) => {
 			return { ...state, importedPlugins: payload };
 		case 'SET_IS_PRO_INSTALLED_DURING_PROCESS':
 			return { ...state, isProInstalledDuringProcess: payload };
+		case 'SET_ACTIONTYPE':
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/context/import-context/import-context-reducer.js:23]: Done

File: core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js [hunk 10]
```diff
@@ -39,6 +41,20 @@ export default function ImportProcess() {
 				setShowUnfilteredFilesDialog( true );
 			}
 		},
+		setCptFromKitLibrary = () => {
+		const cpt = kitState?.data?.manifest[ 'custom-post-type-title' ] || importContext?.data?.uploadedData?.manifest[ 'custom-post-type-title' ];
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js:45]: Removed unnecessary questions mark, if you think I can remove even more please tell me.

File: core/app/modules/import-export/module.php [hunk 11]
```diff
@@ -194,6 +197,12 @@ private function import_stage_1() {
 			throw new \Error( self::MANIFEST_ERROR_KEY );
 		}
 
+		//Check that the user have install_plugins permissions.
+
+		if ( $manifest_data['plugins'] && ! current_user_can( 'install_plugins' ) ) {
+			throw new \Error( self::PERMISSIONS_ERROR_KEY );
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/module.php:203]: Can you please explain?

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/hooks/use-import-kit-library-plugins.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+		{ pluginsData } = usePluginsData( response.data ),
+		{ importPluginsData } = useImportPluginsData( plugins, pluginsData ),
+		{ missing } = importPluginsData || {},
+		handleRequiredPlugins = () => {
+			if ( missing?.length ) {
+				// Saving globally the plugins data that the kit requires in order to work properly.
+				importContext.dispatch( { type: 'SET_PLUGINS', payload: missing } );
+				navigate( '/import/plugins-activation' );
+			}
+		};
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/hooks/use-import-kit-library-plugins.js:22]: This file was removed, I wanted to go through it with Yossi, that is why I left it in the PR.
This file was deleted.

File: core/app/modules/import-export/module.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (StyleShit) [reply core/app/modules/import-export/module.php:203]: https://stackoverflow.com/a/64072873/3829526

File: core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+export default function ImportKitLibraryPlugins() {
+	const importContext = useContext( ImportContext ),
+		plugins = importContext.data.plugins || {},
+		navigate = useNavigate(),
+		{ response } = usePlugins(),
+		//all installed plugins
+		{ pluginsData } = usePluginsData( response.data ),
+		{ importPluginsData } = useImportPluginsData( plugins, pluginsData ),
+		{ missing } = importPluginsData || {},
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-plugins/components/import-kit-library-plugins/import-kit-library-plugins.js:17]: The `missing` name is used all across the import/export app.
Maybe we should fix it in a dedicated patch.

File: core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js [hunk 14]
```diff
@@ -116,6 +140,10 @@ export default function ImportProcess() {
 	return (
 		<Layout type="import">
 			<section>
+				{ plugins?.length > 0 && 0 === importedPlugins.length &&
+					<ImportKitLibraryPlugins />
```
[the remark this answers is not in the document]
Comment (davseve) [reply core/app/modules/import-export/assets/js/pages/import/import-process/import-process.js:144]: Refactored

File: core/app/modules/import-export/assets/js/context/import-context/import-context-provider.js [hunk 15]
```diff
@@ -24,6 +27,10 @@ export default function ImportContextProvider( props ) {
 	);
 }
 
+export function shouldResolveConflict() {
```
Comment (yossi-elementor) [inline core/app/modules/import-export/assets/js/context/import-context/import-context-provider.js:30]: Remove

File: core/app/modules/import-export/assets/js/pages/import/import-kit/hooks/use-import-kit-library-apply-all-plugins.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+export function useImportKitLibraryApplyAllPlugins( plugins ) {
+	const [ missingPlugins, setMissingPlugins ] = useState(),
+		{ response } = usePlugins(),
+		{ pluginsData } = usePluginsData( response.data ),
+		{ importPluginsData } = useImportPluginsData( plugins, pluginsData ),
+		{ missing } = importPluginsData || {};
+
+	useEffect( () => {
+		if ( ! plugins.length ) {
```
Comment (yossi-elementor) [inline core/app/modules/import-export/assets/js/pages/import/import-kit/hooks/use-import-kit-library-apply-all-plugins.js:14]: `plugins && ! plugins.length`

**Review by yossi-elementor — CHANGES_REQUESTED** [review]

**Review by yossi-elementor — APPROVED** [review]

**Pull request merged**